ddtrace/span.py

```diff
@@ -277,6 +280,13 @@ def _finish_ns(self, finish_time_ns):
         for cb in self._on_finish_callbacks:
             cb(self)
 
+    def _override_sampling_decision(self, decision):
+        self.context.sampling_priority = decision
+        set_sampling_decision_maker(self.context, SamplingMechanism.MANUAL)
+        for key in (SAMPLING_RULE_DECISION, SAMPLING_AGENT_DECISION, SAMPLING_LIMIT_DECISION):
+            if key in self._local_root._metrics:
+                del self._local_root._metrics[key]
```

Comment:
Regarding performance, could it be that we're basically always deleting 3 tags from every span but the parent span? Is that expensive enough that it could cause the performance issue?